### ISSUE TITLE
Performance regression on macOS

### ISSUE THREAD

**Issue by geuzaine** [fltk/fltk#127]
The changes in e5b518987007fe2f83394e6d1ba4d8f642b68676 lead to a severe performance regression in Gmsh (http://gmsh.info) on macOS Catalina: updating the message browser is sluggish, as is e.g. scrolling or selecting text.

I failed to reproduce the severity of the performance drop with a simple example, but if you e.g. launch the fltk/test/browser demo with 0da41da713ff5674207334a7c205c958364c18aa and e5b518987007fe2f83394e6d1ba4d8f642b68676 on macOS Catalina, you should "feel" the difference e.g. when selecting multiple lines of text.

**Comment by Albrecht-S** [comment]
Thanks for the report.

@ManoloFLTK ?

**Comment by ManoloFLTK** [comment]
Support of the future macOS 11.0 "Big Sur" forces to change the drawing code of FLTK
because version  0da41da  just does not work under 11.0 public beta.

It's possible to have these changes operate only when macOS 11.0 runs,
that's in attached file (renamed Fl_cocoa.mm.txt to allow upload in this comment, 
just remove the .txt suffix). The code becomes more complex because there are
with it 3 drawing paths
- when macOS < 10.14
- for macOS 10.14 and 10.15
- for macOS 11.0
instead of 2 with  e5b5189 .

Before committing that modified source, I'd like to be sure of the origin of the performance 
regression @geuzaine reports. I see myself no regression with  e5b5189 when running
test/browser on my MacBook Pro and macOS 10.15.6. Is it possible to characterize
in more details a way to reproduce this regression?

Please also report what you get with this modified version of src/Fl_cocoa.mm :
[Fl_cocoa.mm.txt](https://github.com/fltk/fltk/files/5083758/Fl_cocoa.mm.txt)

**Comment by geuzaine** [comment]
Hi @ManoloFLTK,

With the modified version of src/Fl_cocoa.mm (Fl_cocoa.mm.txt) the performance on my mac is back to "normal".

I've tried to replicate the issue on a simpler example, but I did not manage to get something as "bad" as in Gmsh. However when I run fltk/test/browser with the old and new code, I do see a performance difference when I select multiple lines of text: with the old code the selection is instantaneous, whereas with the new version you get a slight lag.

Until I can produce a better example here are two short videos recorded on macOS 10.15.5 on my macbook pro:

 * https://gmsh.info/beta/fast.mov: the "normal" performance of the Gmsh GUI
 * https://gmsh.info/beta/slow.mov: the behaviour with the new "Big Sur" changes

Notice in particular how the progress bar at the bottom updates very slowly (Gmsh uses `Fl::check()` when messages are printed in the log window, so the whole meshing process is actually slowed down - see the displayed CPU time); and how selecting multiple lines of text in the log window at the end of the video is more sluggish with the new version.

(The many `Fl::check()` that Gmsh uses to update the GUI instead of running the GUI in a separate thread are historical: at the time when we designed this we couldn't figure out how to easily distribute statically-linked multi-threaded applications, in particular in combination with OpenMP - which we use to parallelize our meshing algorithms.)
